File: 2024/April/4-9 2073 Time Needed to Buy Tickets/PreCalculation/TimeNeeded.java

class Solution {
    public int timeRequiredToBuy(int[] tickets, int k) {
        int time = 0;
        int length = tickets.length;
        int val = tickets[k];

        for(int i = 0; i < length; i++) {
            if(i <= k) {
                time += Math.min(tickets[i], val);
            }
            else {
                time += Math.min(tickets[i], val - 1);
            }
        }

        return time;
    }
}

File: 2024/April/4-9 2073 Time Needed to Buy Tickets/PreCalculation/TimeNeeded.js
/**
 * @param {number[]} tickets
 * @param {number} k
 * @return {number}
 */
var timeRequiredToBuy = function(tickets, k) {
    var time = 0;
    var length = tickets.length;
    var val = tickets[k];

    for(let i = 0; i < length; i++) {
        if(i <= k) {
            time += Math.min(tickets[i], val);
        }
        else {
            time += Math.min(tickets[i], val - 1);
        }
    }

    return time;
};